### PR TITLE
Extensibility section updated

### PR DESCRIPTION
**Pull request by nedmsmith**
Added text to extensibility section - see issue #10

### CODE REVIEW

File: draft-ietf-rats-corim.md [hunk 1]
```diff
@@ -1194,7 +1195,22 @@ applies to all measurements in the triple, including those in `measurement-value
 
 ## Extensibility {#sec-extensibility}
 
-[^issue] https://github.com/ietf-rats-wg/draft-ietf-rats-corim/issues/10
+CoRIM is described using CDDL {{-cddl}} which includes extensibility support for "sockets" (a.k.a., map structures)
```
Comment (henkbirkholz) [inline draft-ietf-rats-corim.md:1198]: aka "extension points" for type-choices or group-choices

File: draft-ietf-rats-corim.md [hunk 2]
```diff
@@ -1194,7 +1195,22 @@ applies to all measurements in the triple, including those in `measurement-value
...
+and data types. CDDL socket extensions have the form `($$NAME-extension)` where "NAME" is the name of the map.
```
Comment (henkbirkholz) [inline draft-ietf-rats-corim.md:1199]: $$foo is a an extension point for a group-choice
$bar is an extension point for a type choice

Not sure if that level of detail is relevant here, but that is less ambiguous to me.

File: draft-ietf-rats-corim.md [hunk 3]
```diff
@@ -1194,7 +1195,22 @@ applies to all measurements in the triple, including those in `measurement-value
...
+CDDL socket extensibility enables a CoRIM profile to extend the standard definition.
+Typically, map extension requires a convention for code point naming that avoids code-point resue.
```
Comment (henkbirkholz) [inline draft-ietf-rats-corim.md:1201]:
```suggestion
Typically, map extension requires a convention for code point naming that avoids code-point reuse.
```

File: draft-ietf-rats-corim.md [hunk 4]
```diff
@@ -1194,7 +1195,22 @@ applies to all measurements in the triple, including those in `measurement-value
...
+Well-known code points may be in a registry, such as CoSWID {{-coswid-reg}}.
+Additionally, a range of code points may be reserved for vendor-specific use such as negative integers.
+
+Data type extensibility has the form `($NAME-type-choice)` where "NAME" is the type name and '$' signifies type extensibility.
```
Comment (henkbirkholz) [inline draft-ietf-rats-corim.md:1205]: Ah. Here we go. I'd move the referrals and definitions to type- and group-choice to the top. And come back to it here.

File: draft-ietf-rats-corim.md (hunk 2 above)
Comment (yogeshbdeshpande) [reply draft-ietf-rats-corim.md:1199]: @henkbirkholz to work on this PR

File: draft-ietf-rats-corim.md (hunk 1 above)
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1198]:
```suggestion
CoRIM is described using CDDL {{-cddl}} which includes extensibility support for "extension points" (a.k.a., map structures)
```

File: draft-ietf-rats-corim.md (hunk 2 above)
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1199]:
```suggestion
and data types. CDDL map extension points have the form `($$NAME-extension)` where "NAME" is the name of the map and '$$' signifies map extensibility.
```

File: draft-ietf-rats-corim.md [hunk 5]
```diff
@@ -1194,7 +1195,22 @@ applies to all measurements in the triple, including those in `measurement-value
 
 ## Extensibility {#sec-extensibility}
 
-[^issue] https://github.com/ietf-rats-wg/draft-ietf-rats-corim/issues/10
+CoRIM is described using CDDL {{-cddl}} which includes extensibility support for "sockets" (a.k.a., map structures)
+and data types. CDDL socket extensions have the form `($$NAME-extension)` where "NAME" is the name of the map.
+CDDL socket extensibility enables a CoRIM profile to extend the standard definition.
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1200]:
```suggestion
CDDL map extensibility enables a CoRIM profile to extend the base CoRIM definition.
```

File: draft-ietf-rats-corim.md [hunk 6]
```diff
@@ -1194,7 +1195,22 @@ applies to all measurements in the triple, including those in `measurement-value
...
+Typically, map extension requires a convention for code point naming that avoids code-point reuse.
+Well-known code points may be in a registry, such as CoSWID {{-coswid-reg}}.
+Additionally, a range of code points may be reserved for vendor-specific use such as negative integers.
+
+Data type extensibility has the form `($NAME-type-choice)` where "NAME" is the type name and '$' signifies type extensibility.
+
+Socket and type extensions are permitted by the CoRIM schema where the above CDDL extensibility points are defined.
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1207]:
```suggestion
Map and type extensions are permitted by the CoRIM schema where the above CDDL extensibility points are defined.
```

File: draft-ietf-rats-corim.md [hunk 7]
```diff
@@ -1239,7 +1240,22 @@ applies to all measurements in the triple, including those in `measurement-value
 
 ## Extensibility {#sec-extensibility}
 
-[^issue] https://github.com/ietf-rats-wg/draft-ietf-rats-corim/issues/10
+CoRIM is described using CDDL {{-cddl}} which includes extensibility support for "extension points" (a.k.a., map structures)
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1243]: The whole structure of the section needs reorganising. I will suggest a better rewrite as an edit to this section!

File: draft-ietf-rats-corim.md [hunk 8]
```diff
@@ -1239,7 +1240,22 @@ applies to all measurements in the triple, including those in `measurement-value
...
+
+Data type extensibility has the form `($NAME-type-choice)` where "NAME" is the type name and '$' signifies type extensibility.
+
+Map and type extensions are permitted by the CoRIM schema where the above CDDL extensibility points are defined.
+Schema extensions should be documented to facilitate interoperability. CoRIM profiles are best used to documennt
+vendor or industry defined extensions.
+
+This specification uses CDDL type extension. Type extensions contained in this specification are considered
+as "standard CoRIM types". Standard CoRIM types, along with the other CDDL expressions in this specification,
+set the baseline for CoRIM interoperability.
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1258]:
```suggestion
The base CORIM schema is described using CDDL {{-cddl}} that can be extended 
only at specific allowed points known as "extension points"

The following types of extensions are supported in CoRIM

## Map Extensions
Map Extensions provides extensibility support to CoRIM Map structures.
CDDL map extensibility enables a CoRIM profile to extend the base CoRIM definition.
CDDL map extension points have the form `($$NAME-extension)` where "NAME" is the name of the map
and '$$' signifies map extensibility. Typically, map extension requires a convention 
for code point naming that avoids code-point reuse.
Well-known code points may be in a registry, such as CoSWID {{-coswid-reg}}.
Additionally, a range of code points may be reserved for vendor-specific use such as negative integers.

## Data Type Extensions

Data type extensibility has the form `($NAME-type-choice)` where "NAME" is the type name 
and '$' signifies type extensibility.

Schema extensions (Map or Data Type) should be documented to facilitate interoperability. CoRIM profiles are best used to documennt
vendor or industry defined extensions.
```

**Review by yogeshbdeshpande — APPROVED** [review]
LGTM!

**Review by henkbirkholz — APPROVED** [review]
will add CDDLlingo to extension types later

**Review by andrew-draper — APPROVED** [review]

**Pull request merged**